src/tiledb/cloud/client.py

```diff
@@ -672,13 +672,15 @@ def __init__(
     @property
     def _client_v1(self):
         if not self.__client_v1:
-            self.retry_mode(self._mode)
+            self._retry_mode(self._mode)
+            self._rebuild_clients()
         return self.__client_v1
 
     @property
     def _client_v2(self):
         if not self.__client_v2:
-            self.retry_mode(self._mode)
+            self._retry_mode(self._mode)
+            self._rebuild_clients()
```

Comment:
For clarity, call the methods that were called in the constructor originally, instead of rebuilding as a side effect of setting the retry mode.